shell/browser/lib/bluetooth_chooser.cc

```diff
@@ -105,38 +93,42 @@ void BluetoothChooser::AddOrUpdateDevice(const std::string& device_id,
                                          bool is_gatt_connected,
                                          bool is_paired,
                                          int signal_strength_level) {
-  if (refreshing_) {
-    // If the list of bluetooth devices is currently being generated don't fire
-    // an event
+  // Don't fire an event during refresh.
+  if (refreshing_)
     return;
-  }
-
-  auto [iter, changed] = device_map_.try_emplace(device_id, device_name);
-  if (!changed && should_update_name) {
-    iter->second = device_name;
-    changed = true;
-  }
 
+  // Emit a select-bluetooth-device handler to allow for user to listen for
+  // bluetooth device found. If there's no listener in place, then select the
+  // first device that matches the filters provided.
+  auto [iter, changed] =
+      device_id_to_name_map_.try_emplace(device_id, device_name);
```

Comment:
Looks like the `if (!changed && should_update_name)` got lost here, which seems like a bug?

Reply:
yep, fixed! lost in the shuffle